### PR TITLE
Powerbi sso configuration

### PR DESCRIPTION
**Pull request by jreynard-code**
Dev Review rules:
- 3 PR max
- at the second feedback, prefer meet instead of github conversation

Definition of Done:
- [ ] Testing
- [X] No error logs in web console (except iFrame errors)
- [X] Create or Update documentation (README file for users/integrators)
- [ ] Check the [test page](https://spaceport.cosmotech.com/confluence/pages/viewpage.action?spaceKey=QA&title=Azure+Sample+Webapp) in order to manually validate the feature 
- [X] Clean code (regroup configuration variable, typos, no console.log...)

_______________________

Define two constants to handle "real" SSO with Power BI.
Until now, connection to PowerBI was done through a service principal.
Now, you can chose the authentication method:
- continue to use a service account (by extension, continue to use GetEmbedInfo azure function and do not have the possibility to adapt reports data regarding connected user)
- use User account (and do have the possibility to adapt reports data regarding connected user, via RLS in PowerBI)

### CODE REVIEW

File: src/state/sagas/powerbi/GetPowerBIEmbedInfo/GetPowerBIEmbedInfoData.js [hunk 1]
```diff
@@ -1,50 +1,61 @@
...
-      if (data.error) {
+      let accesses, error;
+      if (USE_POWER_BI_WITH_USER_CREDENTIALS) {
+        const response = yield PowerBIService.getPowerBIData();
+        accesses = response?.accesses;
+        error = response?.error;
+      } else {
+        const { data } = yield clientApi.get(GET_EMBED_INFO_URL);
+        accesses = data?.accesses;
+        error = data?.error;
```
[the remark this answers is not in the document]
Comment (jreynard-code) [reply src/state/sagas/powerbi/GetPowerBIEmbedInfo/GetPowerBIEmbedInfoData.js:31]: Unfortunately, it's not possible :s

File: src/config/AppInstance.js [hunk 2]
```diff
@@ -13,3 +13,8 @@ export const DEFAULT_BASE_PATH = 'https://dev.api.cosmotech.com';
 // Cosmo solution identifiers
 export const ORGANIZATION_ID = 'O-gZYpnd27G7';
 export const WORKSPACE_ID = 'W-rXeBwRa0PM';
+
+// Power BI information
+// true => Aad false => Embed
+export const USE_POWER_BI_WITH_USER_CREDENTIALS = false;
+export const POWER_BI_WORKSPACE_ID = '290de699-9026-42c0-8c83-e4e87c3f22dd';
```
[the remark this answers is not in the document]
Comment (jreynard-code) [reply src/config/AppInstance.js:20]: As discussed: 

- USE_POWER_BI_WITH_USER_CREDENTIALS => AppConfiguration.js
- POWER_BI_WORKSPACE_ID => AppInstance.js

File: doc/appInstance.md [hunk 3]
```diff
@@ -34,3 +34,9 @@ Steps to enable Application Insights in the webapp:
 - then, edit the configuration file _src/config/AppInstance.js_ in the source code of your webapp:
   - replace the value of APPLICATION_INSIGHTS_INSTRUMENTATION_KEY with the instrumentation key that has been
     generated for your application
+  
+### Power BI
+
+Parameter to define Power BI usages:
+- **POWER_BI_WORKSPACE_ID** if _USE_POWER_BI_WITH_USER_CREDENTIALS_ (defines in [src/config/AppInstance.js](../src/config/AppInstance.js)) is set to `true`, it represents the Power BI workspace ID used
```
Comment (csm-thu) [inline doc/appInstance.md:41]: _defined_*

File: src/config/AppConfiguration.js [hunk 4]
```diff
@@ -35,3 +35,7 @@ export const ADD_SCENARIO_PARENT_LAST_RUN_ID_PARAMETER = false;
 export const ADD_SCENARIO_MASTER_ID_PARAMETER = false;
 export const ADD_SCENARIO_MASTER_LAST_RUN_ID_PARAMETER = false;
 export const ADD_SCENARIO_RUN_TEMPLATE_NAME_PARAMETER = false;
+
+// Power BI
+// true => Aad false => Embed
```
Comment (csm-thu) [inline src/config/AppConfiguration.js:40]: Can we remove the `true => Aad false => Embed` comment (we have doc to explain that) or put it on the same line as the parameter, to decrease the risks of conflicts for solutions in the configuration files ?

**Review by csm-thu — APPROVED** [review]
Some improvements suggestions, but LGTM :+1:

**Pull request merged**